docs/02-architecture/06-client-side-architecture.md

```diff
@@ -0,0 +1,285 @@
+
+*Note that if you want to use config in a JS module, you shouldn't use the window object, but include it directly in your module via the config.js utility.*
+
+#### applyRenderConditions
+
+Choose how the browser should render the page before any painting begins. [applyRenderConditions.js](https://github.com/guardian/frontend/blob/master/common/app/templates/inlineJS/blocking/applyRenderConditions.scala.js).
+
+Applies classes to the document based on support including svg, flexbox and replaces `js-off` with `js-on`.
+
+#### loadFonts
+
+Described at the top of the loadFonts.scala.js:
+
+**bypass normal browser font-loading to avoid the FOIT.**
+
+works like this:
+
+do you have fonts in localStorage?
+
+- **yes** – inject them (minimises 2nd layout as fonts are loaded before DOM is created)
+- **no**  – did the localStorage check go ok?
+	- **yes** – ajax them in as JSON immediately, inject them and save them to localStorage
+	- **no**  – load font files async using @@font-face
+	
+#### Enable non-blocking stylesheets
+
+A util that borrows heavily from [loadCSS](https://github.com/filamentgroup/loadCSS), it loads CSS async so that non-critical CSS doesn't block rendering.
+
+#### Load the main app async
+
+We polyfill the `async` attribute to prevent parser blocking by creating a script element and inserting it into the page. This is the main app.js which contains standard and enhanced JS.
+
+*Note that in Dev we get curl and require [boot.js](https://github.com/guardian/frontend/blob/master/static/src/javascripts/boot.js) immediately here but on Prod we concatenate curl and boot.js with bootstraps/standard*
+
+*Double Note: in head.scala.html we use a link element with prefetch attr and href set to app.js so that we head off and fetch it as early as possible*
+
+`<link rel="prefetch" href="@Static("javascripts/app.js")">`
+
+#### Cloudwatch beacon
+
+This is where we would ping cloudwatch.. if we had anything to ping about..
+
+### Inline non-blocking JS
+
+#### getUserData
+
+The [getUserData](https://github.com/guardian/frontend/blob/master/common/app/templates/inlineJS/nonBlocking/getUserData.scala.js) util puts the current logged-in user's data in the JS config ready to be used by other JS.
+
+#### detectAdblock
+
+Secrets.
+
+#### showUserName
+
+[Puts the user's username](https://github.com/guardian/frontend/blob/master/common/app/templates/inlineJS/nonBlocking/showUserName.scala.js) into the header (increases perceived rendering speed as their username is there before the app.js has to download and parse).
+
+#### Ophan config
+
+Gets the [Ophan browserId](https://github.com/guardian/frontend/blob/master/common/app/templates/inlineJS/nonBlocking/ophanConfig.scala.js) which is used across analytics to tie data together.
+
+### Bootstraps
+
+In javascripts/bootstraps we define all the entry points for each bundle described in [requirejs.js](). 
```

Comment:
Ah yeah, whoops, did the last bit of this on the train and forgot I needed to link a few things!